src/Avalonia.Controls/TextBox.cs

```diff
@@ -145,7 +145,7 @@ public class TextBox : TemplatedControl, UndoRedoHelper<TextBox.UndoRedoState>.I
                 (o, v) => o.UndoLimit = v,
                 unsetValue: -1);
 
-        struct UndoRedoState : IEquatable<UndoRedoState>
+        readonly struct UndoRedoState : IEquatable<UndoRedoState>
```

Comment:
I have an unrelated question - does anybody know if that `Equals` implementation below makes sense? Or is `CaretPosition` supposed to be ignored?

Reply:
From my understanding, we need to ignore CaretPosition, so new snapshot won't be created when only caret position was changed.